tools/pylint-extensions/pylint-guidelines-checker/pylint_guidelines_checker.py

```diff
@@ -2028,6 +2028,47 @@ def _check_import(self, name, node):
                 msgid="no-raise-with-traceback", node=node, confidence=None
             )
 
+class DeleteOperationReturnStatement(BaseChecker):
+    __implements__ = IAstroidChecker
+
+    """Rule to check that a delete* or begin_delete* returns None or LROPoller[None]."""
+    name = "delete-operation-wrong-return-type"
+    priority = -1
+    msgs = {
+        "C4752": (
+            "delete* or begin_delete* does not return None or LROPoller[None].",
+            "delete-operation-wrong-return-type",
+            "delete* or begin_delete* functions should return None or LROPoller[None]."
+        ),
+    }
+
+    def visit_functiondef(self,node):
+        """Visits all delete functions and checks that their return types
+        are LROPoller or None. """
+        try:
+            if node.returns.as_string() == "None":
+                # If there are residual comment typehints or no return value,
+                # we dont want to throw an error
+                return
+            if node.name.startswith("delete") and node.parent.name.endswith("Client"):
+                if node.returns.as_string() != "None":
```

Comment:
does `node.returns` access the type hints?

Reply:
it accesses the return annotation, there is also type_comment_returns that accesses the return type from a type comment.